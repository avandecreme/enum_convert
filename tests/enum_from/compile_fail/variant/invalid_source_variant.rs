use enum_convert::EnumFrom;

enum Source {
    Unit,
}

#[derive(EnumFrom)]
#[enum_from(Source)]
enum Target {
    #[enum_from(Source::NonExistent)] // Invalid variant name
    Unit,
}

fn main() {}
