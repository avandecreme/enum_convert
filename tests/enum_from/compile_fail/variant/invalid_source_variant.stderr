error[E0599]: no variant or associated item named `NonExistent` found for enum `Source` in the current scope
  --> tests/enum_from/compile_fail/variant/invalid_source_variant.rs:10:25
   |
3  | enum Source {
   | ----------- variant or associated item `NonExistent` not found for this enum
...
10 |     #[enum_from(Source::NonExistent)] // Invalid variant name
   |                         ^^^^^^^^^^^ variant or associated item not found in `Source`
